cmd/nonce-service/main.go

```diff
@@ -17,18 +20,69 @@ type config struct {
 		cmd.ServiceConfig
 		Syslog  cmd.SyslogConfig
 		MaxUsed int
+
+		RemoteNonceServices []cmd.GRPCClientConfig
 	}
 }
 
 type nonceServer struct {
-	inner *nonce.NonceService
+	inner          *nonce.NonceService
+	remoteServices []noncepb.NonceServiceClient
+	log            blog.Logger
+}
+
+func (ns *nonceServer) remoteRedeem(ctx context.Context, msg *noncepb.NonceMessage) bool {
+	deadline, ok := ctx.Deadline()
+	if !ok {
+		ns.log.Err("Context passed to remoteRedeem does not have a deadline")
+		return false
+	}
+	subCtx, cancel := context.WithDeadline(ctx, deadline.Add(-time.Millisecond*250))
+	defer cancel()
+	forwarded := true
+	msg.Forwarded = &forwarded
+	results := make(chan bool, len(ns.remoteServices))
+	wg := new(sync.WaitGroup)
+	for _, remote := range ns.remoteServices {
+		wg.Add(1)
+		go func(r noncepb.NonceServiceClient) {
+			defer wg.Done()
+			resp, err := r.Redeem(subCtx, msg)
+			if err != nil {
+				ns.log.Errf("remote Redeem call failed: %s", err)
+				return
+			}
+			results <- *resp.Valid
+		}(remote)
+	}
+	go func() {
+		wg.Wait()
+		close(results)
+	}()
+	for result := range results {
+		select {
+		case <-subCtx.Done():
+			return false
+		default:
+			if result {
+				return true
+			}
+		}
+	}
+	return false
 }
 
-func (ns *nonceServer) Redeem(_ context.Context, msg *noncepb.NonceMessage) (*noncepb.ValidMessage, error) {
+func (ns *nonceServer) Redeem(ctx context.Context, msg *noncepb.NonceMessage) (*noncepb.ValidMessage, error) {
 	if msg.Nonce == nil {
 		return nil, errors.New("Incomplete gRPC request message")
 	}
 	valid := ns.inner.Valid(*msg.Nonce)
+	// If the nonce was not valid, we have configured remote nonce services
```

Comment:
Commas to clarify:

	// If the nonce was not valid, we have configured remote nonce services,
	// and this Redeem message wasn't forwarded, then forward it to the
	// remote services